storage/remote/queue_manager.go

```diff
@@ -827,19 +834,18 @@ func (t *QueueManager) calculateDesiredShards() int {
 	// When behind we will try to catch up on a proporation of samples per tick.
 	// This works similarly to an integral accumulator in that pending samples
 	// is the result of the error integral.
-	const integralGain = 0.1 / float64(shardUpdateDuration/time.Second)
+	const integralGain = 0.5 / float64(shardUpdateDuration/time.Second)
 
 	var (
 		timePerSample = dataOutDuration / dataOutRate
-		desiredShards = timePerSample * (dataInRate*dataKeptRatio + integralGain*dataPending)
+		desiredShards = timePerSample * (dataInRate*dataKeptRatio + integralGain*dataPendingRate)
```

Comment:
The units here no longer make sense, `integralGain` has units of `1/seconds` so if multiplied by a rate we are now trying to add `1/seconds` and `1/seconds^2`.

Reply:
https://github.com/prometheus/prometheus/blob/752c4f11ae86effa9a46f017f2feb66730c67ed8/storage/remote/queue_manager.go#L830

hmm, cmiiw `integralGain` has no units (`const * T/T` => `const * const`)

Though I found another issue in `dataOutDuration` which I have now fixed.

Reply:
Hmm... I could also be wrong. I think `integralGain` is `const / (nanoseconds/(nanoseconds/seconds)) => const / seconds` right now though?

Reply:
>I think integralGain is const / (nanoseconds/(nanoseconds/seconds)) => const / seconds right now

why does `time.Second` have units `(nanoseconds/seconds)`? 🤔 

here's what I have in my head:

```
| var             | formula                                        | dimensions                                     |
|-----------------+------------------------------------------------+------------------------------------------------|
| dataInRate      | rate of samples ingested per sec               | samples/T                                      |
| dataOutRate     | rate of samples output per sec                 | samples/T                                      |
| dataDroppedRate | rate of samples dropped per sec                | samples/T                                      |
| dataOutDuration | rate of time spent on each send per sec        | T/T => const                                   |
| dataKeptRatio   | 1 - (dataDroppedRate / dataInRate)             | 1 - (samples/T) / (samples/T) => const         |
| dataPendingRate | dataInRate - dataOutRate - dataDroppedRate     | samples/T - samples/T - samples/T => samples/T |
| integralGain    | 0.5 / float64(shardUpdateDuration/time.Second) | const / (T/T) => const                         |
| timePerSample   | dataOutDuration / dataOutRate                  | const / (samples/T) => T/samples               |
```

Thus, we get `desiredShards` which has the formula:

`timePerSample * (dataInRate*dataKeptRatio + integralGain*dataPendingRate)` become:

```
T/samples * (samples/T * const + const * samples/T)
T/samples * (samples/T)
const
```

Reply:
It is because any `time.Duration` in Go is stored in nanoseconds. `time.Second` is then used to convert from nanoseconds to seconds to be in the same unit as all the other rates we have.

Reply:
agree, but that its stored in ns is an implementation detail and `myTimeDuration.Second()` conversion is just division by const `1e9`, isn't it? Note, `time.Second` is just a [const](https://github.com/golang/go/blob/1d91551b7326383343c7c143a8ac299d0a685289/src/time/time.go#L608-L612).

Reply:
> > why does time.Second have units (nanoseconds/seconds)? 🤔

> It is because any time.Duration in Go is stored in nanoseconds. time.Second is then used to convert from nanoseconds to seconds to be in the same unit as all the other rates we have.

Conversion from nanosecond -> seconds shouldn't change the units since it's just division by a const (`1e9`). 
`time.Second` should have dimension `T` still, isn't it?
https://github.com/golang/go/blob/1d91551b7326383343c7c143a8ac299d0a685289/src/time/time.go#L608-L612

Reply:
Hey everyone what's going on